### PR TITLE
Improve adding application docs

### LINKED ISSUE #73: Re-evaluate using overlays/bases for this repo
We now have a situation where one argocd app is deploying manifests for 2 different environments (infra/cnv), but both these environments may want to consume from the same base, which could yield conflicting application names (hence CR names), this is obviously a conflict that cannot occur. This is an inherent flaw with the design of using base/overlays for this purpose, so we may want to re-consider that format for this repo. Thoughts?

### PR DESCRIPTION
**Pull request by HumairAK**
Advances #63, though I suspect we'll need to revisit it again after we resolve #73 - this should at least fix the outdated information/links. I also remove mention of adding things to base for now and simplified the instructions.

### CODE REVIEW

File: docs/add_application.md [hunk 1]
```diff
@@ -1,48 +1,22 @@
 # Adding ArgoCD Applications
 
-This repo contains all operate-first ArgoCD [Applications](https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/#applications).
-
-Please see the use-cases below and follow the instructions that best fit your application's needs.
+This repo contains all Operate First ArgoCD [Applications](https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/#applications).
 
 # Pre-requisites
 - You should have [kustomize](https://kubectl.docs.kubernetes.io/installation/kustomize/) installed
 - You should be familiar with [kustomize ovelrays](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/#bases-and-overlays)
```
Comment (tumido) [inline docs/add_application.md:7]: I know this is not part of your PR, however can you please fix the typo here?
```suggestion
- You should be familiar with [kustomize overlays](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/#bases-and-overlays)
```

**Review by tumido — COMMENTED** [review]
I have one extra wish to ask you to fix one typo I've noticed. Other than that 

/lgtm

**Review by 4n4nd — COMMENTED** [review]
/lgtm

**Review by 4n4nd — APPROVED** [review]

**Review by anishasthana — APPROVED** [review]
/lgtm

**Pull request merged**